### PR TITLE
feat(manager): Implementing Analytics section of Dashboard

### PR DESCRIPTION
**Pull request by ayushsatyam146**
## Closes / Fixes / Resolves
SPASHIP-1119

## Explain the feature/fix
Implemented analytics section of the SPAship dashboard

## Does this PR introduce a breaking change

No

## Screenshot(s)

<details>
<summary>View Screenshots</summary>

image
image
image

</details>

### CODE REVIEW

**Comment by akhilmhdh** [comment]
hey @ayushsatyam146 kindly rebase ur codebase

File: packages/api/src/server/analytics/service/analytics.factory.ts [hunk 1]
```diff
@@ -78,4 +111,25 @@ export class AnalyticsFactory {
...
+    return [startDate, endDate];
+  }
+
+  groupEphemeral(response: Object, obj: any) {
+    const result = response[AnalyticsFactory.ephemeral].findIndex((element) => element.startDate === obj.startDate);
+    if (result !== -1) response[AnalyticsFactory.ephemeral][result].count += obj.count;
+    else {
+      obj.env = AnalyticsFactory.ephemeral;
+      response[AnalyticsFactory.ephemeral].push(obj);
+    }
```
Comment (hybridx) [inline packages/api/src/server/analytics/service/analytics.factory.ts:128]:
```suggestion
    if (result === -1) {
    obj.env = AnalyticsFactory.ephemeral;
      response[AnalyticsFactory.ephemeral].push(obj);
      return;
    }
    response[AnalyticsFactory.ephemeral][result].count += obj.count;
```
can we do something like this?

File: packages/api/src/server/analytics/service/analytics.service.ts [hunk 2]
```diff
@@ -13,10 +15,14 @@ export class AnalyticsService {
 
   private static readonly channel: string = 'activity_stream';
 
+  private static readonly ephemeral: string = 'ephemeral';
+
   private static readonly defaultSkip: number = 0;
 
   private static readonly defaultLimit: number = 100;
 
+  private static readonly defaultDays: number = 120;
```
Comment (hybridx) [inline packages/api/src/server/analytics/service/analytics.service.ts:24]: @ayushsatyam146 @soumyadip007 Should we change all of these variables to capital?

File: packages/manager/package.json [hunk 3]
```diff
@@ -50,22 +50,23 @@
     "axios": "0.27.2",
     "dayjs": "1.11.7",
     "next": "12.3.4",
-    "next-auth": "4.18.0",
+    "next-auth": "4.18.4",
```
Comment (hybridx) [inline packages/manager/package.json:53]: Check if there are any breaking changes @ayushsatyam146

**Review by hybridx — CHANGES_REQUESTED** [review]

**Pull request closed without merging**